Recent changes in NDData and/or Quantity causes test failuere in test_aperture_photometry

This should be trivial to fix, I'll deal with it in the next couple of days.

```
=================================== FAILURES ===================================
___________ TestInputNDData.test_basic_circular_aperture_photometry ____________

self = <photutils.tests.test_aperture_photometry.TestInputNDData object at 0x1b63e230>

    def test_basic_circular_aperture_photometry(self):
    
        aperture = CircularAperture(self.position, self.radius)
        table = aperture_photometry(self.data, aperture,
>                                   method='exact')

../photutils/photutils/tests/test_aperture_photometry.py:416: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

data = NDData([[ 1.,  1.,  1., ...,  1.,  1.,  1.],
        [ 1.,  1.,  1., ...,  1.,....,  1.,  1., ...,  1.,  1.,  1.],
        [ 1.,  1.,  1., ...,  1.,  1.,  1.]])
apertures = <photutils.aperture_core.CircularAperture object at 0x1b63ee10>
unit = None, wcs = None, error = None, effective_gain = None, mask = None
method = 'exact', subpixels = 5, pixelwise_error = True

    def aperture_photometry(data, apertures, unit=None, wcs=None, error=None,
                            effective_gain=None, mask=None, method='exact',
                            subpixels=5, pixelwise_error=True):
        """
        Sum flux within an aperture at the given position(s).
    
        Parameters
        ----------
        data : array_like, `~astropy.io.fits.ImageHDU`, `~astropy.io.fits.HDUList`
            The 2-d array on which to perform photometry. ``data`` should be
            background-subtracted.  Units are used during the photometry,
            either provided along with the data array, or stored in the
            header keyword ``'BUNIT'``.
        apertures : `~photutils.Aperture` instance
            The apertures to use for the photometry.
        unit : `~astropy.units.UnitBase` instance, str
            An object that represents the unit associated with ``data``.  Must
            be an `~astropy.units.UnitBase` object or a string parseable by the
            :mod:`~astropy.units` package. An error is raised if ``data``
            already has a different unit.
        wcs : `~astropy.wcs.WCS`, optional
            Use this as the wcs transformation. It overrides any wcs transformation
            passed along with ``data`` either in the header or in an attribute.
        error : float or array_like, optional
            Error in each pixel, interpreted as Gaussian 1-sigma uncertainty.
        effective_gain : float or array_like, optional
            Ratio of counts (e.g., electrons or photons) to units of the
            data (e.g., ADU), for the purpose of calculating Poisson error
            from the object itself. If ``effective_gain`` is `None`
            (default), ``error`` is assumed to include all uncertainty in
            each pixel. If ``effective_gain`` is given, ``error`` is assumed
            to be the "background error" only (not accounting for Poisson
            error in the flux in the apertures).
        mask : array_like (bool), optional
            Mask to apply to the data.  Masked pixels are excluded/ignored.
        method : str, optional
            Method to use for determining overlap between the aperture and pixels.
            Options include ['center', 'subpixel', 'exact'], but not all options
            are available for all types of apertures. More precise methods will
            generally be slower.
    
            * ``'center'``
                A pixel is considered to be entirely in or out of the aperture
                depending on whether its center is in or out of the aperture.
            * ``'subpixel'``
                A pixel is divided into subpixels and the center of each
                subpixel is tested (as above). With ``subpixels`` set to 1, this
                method is equivalent to 'center'. Note that for subpixel
                sampling, the input array is only resampled once for each
                object.
            * ``'exact'`` (default)
                The exact overlap between the aperture and each pixel is
                calculated.
        subpixels : int, optional
            For the ``'subpixel'`` method, resample pixels by this factor (in
            each dimension). That is, each pixel is divided into
            ``subpixels ** 2`` subpixels.
        pixelwise_error : bool, optional
            For ``error`` and/or ``effective_gain`` arrays. If `True`,
            assume ``error`` and/or ``effective_gain`` vary significantly
            within an aperture: sum contribution from each pixel. If
            `False`, assume ``error`` and ``effective_gain`` do not vary
            significantly within an aperture. Use the single value of
            ``error`` and/or ``effective_gain`` at the center of each
            aperture as the value for the entire aperture.  Default is
            `True`.
    
        Returns
        -------
        phot_table : `~astropy.table.Table`
            A table of the photometry with the following columns:
    
            * ``'aperture_sum'``: Sum of the values within the aperture.
            * ``'aperture_sum_err'``: Corresponding uncertainty in
              ``'aperture_sum'`` values.  Returned only if input ``error``
              is not `None`.
            * ``'xcenter'``, ``'ycenter'``: x and y pixel coordinates of the
              center of the apertures. Unit is pixel.
            * ``'xcenter_input'``, ``'ycenter_input'``: input x and y
              coordinates as they were given in the input ``positions``
              parameter.
    
            The metadata of the table stores the version numbers of both astropy
            and photutils, as well as the calling arguments.
        """
        dataunit = None
        datamask = None
        wcs_transformation = wcs
    
        if isinstance(data, (fits.PrimaryHDU, fits.ImageHDU)):
            header = data.header
            data = data.data
    
            if 'BUNIT' in header:
                dataunit = header['BUNIT']
    
            # TODO check how a mask can be stored in the header, it seems like
            # full pixel masks are not supported by the FITS standard, look for
            # real life examples (e.g. header value stores the fits number of
            # fits extension where the pixelmask is stored?)
            if 'MASK' in header:
                datamask = header.mask
    
        elif isinstance(data, fits.HDUList):
            # TODO: do it in a 2d array, and thus get the light curves as a
            # side-product? Although it's not usual to store time series as
            # HDUList
    
            for i in range(len(data)):
                if data[i].data is not None:
                    warnings.warn("Input data is a HDUList object, photometry is "
                                  "only run for the {0}. HDU."
                                  .format(i), AstropyUserWarning)
                    return aperture_photometry(data[i], apertures, unit, wcs,
                                               error, effective_gain, mask,
                                               method, subpixels, pixelwise_error)
    
        # this is basically for NDData inputs and alike
        elif hasattr(data, 'data') and not isinstance(data, np.ndarray):
            if data.wcs is not None and wcs_transformation is None:
                wcs_transformation = data.wcs
            datamask = data.mask
    
        if hasattr(data, 'unit'):
            dataunit = data.unit
    
        if unit is not None and dataunit is not None:
            if unit != dataunit:
                raise u.UnitsError('Unit of input data ({0}) and unit given by '
                                   'unit argument ({1}) are not identical.'.
                                   format(dataunit, unit))
            data = u.Quantity(data, unit=dataunit, copy=False)
        elif unit is None:
            if dataunit is not None:
>               data = u.Quantity(data, unit=dataunit, copy=False)

../photutils/photutils/aperture_core.py:1284: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'astropy.units.quantity.Quantity'>
value = array(NDData([[ 1.,  1.,  1., ...,  1.,  1.,  1.],
        [ 1.,  1.,  1., .......,  1.,  1.,  1.],
        [ 1.,  1.,  1., ...,  1.,  1.,  1.]]), dtype=object)
unit = Unit("adu"), dtype = None, copy = False, order = None, subok = False
ndmin = 0

    def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
                subok=False, ndmin=0):
    
        if unit is not None:
            # convert unit first, to avoid multiple string->unit conversions
            unit = Unit(unit)
    
        # optimize speed for Quantity with no dtype given, copy=False
        if isinstance(value, Quantity):
            if unit is not None and unit is not value.unit:
                value = value.to(unit)
                # the above already makes a copy (with float dtype)
                copy = False
    
            if not subok and type(value) is not cls:
                value = value.view(cls)
    
            if dtype is None:
                if not copy:
                    return value
    
                if not np.can_cast(np.float32, value.dtype):
                    dtype = np.float
    
            return np.array(value, dtype=dtype, copy=copy, order=order,
                            subok=True, ndmin=ndmin)
    
        # Maybe list/tuple of Quantity? short-circuit array for speed
        if(not isinstance(value, np.ndarray) and isiterable(value) and
           all(isinstance(v, Quantity) for v in value)):
            if unit is None:
                unit = value[0].unit
            value = [q.to(unit).value for q in value]
            value_unit = unit  # signal below that conversion has been done
            copy = False  # copy already made
    
        else:
            # If the value has a `unit` attribute and if not None
            # (for Columns with uninitialized unit), treat it like a quantity.
            value_unit = getattr(value, 'unit', None)
            if value_unit is None:
                # Default to dimensionless for no (initialized) unit attribute.
                if unit is None:
                    unit = dimensionless_unscaled
                value_unit = unit  # signal below that no conversion is needed
            else:
                try:
                    value_unit = Unit(value_unit)
                except Exception as exc:
                    raise TypeError("The unit attribute {0} of the input could "
                                    "not be parsed as an astropy Unit, raising "
                                    "the following exception:\n{1}"
                                    .format(repr(value.unit), exc))
    
                if unit is None:
                    unit = value_unit
                elif unit is not value_unit:
                    copy = False  # copy will be made in conversion at end
    
        value = np.array(value, dtype=dtype, copy=copy, order=order,
                         subok=False, ndmin=ndmin)
    
        # check that array contains numbers or long int objects
        if (value.dtype.kind in 'OSU' and
            not (value.dtype.kind == 'O' and
                 isinstance(value.item(() if value.ndim == 0 else 0),
                            numbers.Number))):
>           raise TypeError("The value must be a valid Python or "
                            "Numpy numeric type.")
E           TypeError: The value must be a valid Python or Numpy numeric type.

astropy/units/quantity.py:245: TypeError
=============== 1 failed, 416 passed, 3 skipped in 16.84 seconds ===============
```

--- a/photutils/aperture_core.py
+++ b/photutils/aperture_core.py
@@ -1239,9 +1239,11 @@ def aperture_photometry(data, apertures, unit=None, wcs=None, error=None,
 
     Notes
     -----
-    ``aperture_photometry`` is decorated with
-    `~astropy.nddata.support_nddata` when astropy >v1.0 is available.
+    This function is decorated with `~astropy.nddata.support_nddata` and
+    thus supports `~astropy.nddata.NDData` objects as input for Astropy
+    version >=1.0.
     """
+
     dataunit = None
     datamask = None
     wcs_transformation = wcs

--- a/photutils/aperture_core.py
+++ b/photutils/aperture_core.py
@@ -25,6 +25,7 @@ from .utils.wcs_helpers import (skycoord_to_pixel_scale_angle, assert_angle,
 from astropy import __version__ as astropy_version
 if version.LooseVersion(astropy_version) > version.LooseVersion('1.0'):
     from astropy.wcs.utils import skycoord_to_pixel
+    from astropy.nddata import support_nddata
     skycoord_to_pixel_mode = 'all'
 else:
     from .extern.wcs_utils import skycoord_to_pixel
@@ -1147,6 +1148,7 @@ class RectangularAnnulus(PixelAperture):
         return flux
 
 
+@support_nddata
 def aperture_photometry(data, apertures, unit=None, wcs=None, error=None,
                         effective_gain=None, mask=None, method='exact',
                         subpixels=5, pixelwise_error=True):

--- a/photutils/aperture_core.py
+++ b/photutils/aperture_core.py
@@ -31,6 +31,9 @@ else:
     from .extern.wcs_utils import skycoord_to_pixel
     skycoord_to_pixel_mode = 'wcs'
 
+    def support_nddata(_func):
+        return _func
+
 
 __all__ = ['Aperture', 'SkyAperture', 'PixelAperture',
            'SkyCircularAperture', 'CircularAperture',

--- a/photutils/aperture_core.py
+++ b/photutils/aperture_core.py
@@ -1236,6 +1236,11 @@ def aperture_photometry(data, apertures, unit=None, wcs=None, error=None,
 
         The metadata of the table stores the version numbers of both astropy
         and photutils, as well as the calling arguments.
+
+    Notes
+    -----
+    ``aperture_photometry`` is decorated with
+    `~astropy.nddata.support_nddata` when astropy >v1.0 is available.
     """
     dataunit = None
     datamask = None

